gcc/rust/ast/rust-ast-dump.cc

```diff
@@ -399,23 +474,51 @@ Dump::visit (CompoundAssignmentExpr &expr)
 
 void
 Dump::visit (GroupedExpr &expr)
-{}
+{
+  stream << '(';
+  expr.get_expr_in_parens ()->accept_vis (*this);
+  stream << ')';
+}
 
 void
 Dump::visit (ArrayElemsValues &elems)
-{}
+{
+  auto &vals = elems.get_values ();
+  if (vals.size () >= 1)
```

Comment:
This is a recurring pattern which is not very nice to read. I think we should make a `emit_list_without_trailing_comma` method. I can add it in some next PR.

Reply:
That's a very good idea! Feel free to do so, that would be a nice refactor. You'll probably need to use templates